webapp/model/Dispute.php

```diff
@@ -163,18 +168,21 @@ private function agentIsNotInParty($agent, $party) {
 
     public function canBeViewedBy($loginID) {
         $account = AccountDetails::getAccountById($loginID);
-        $viewableDisputes = $account->getAllDisputes();
-        foreach($viewableDisputes as $dispute) {
-            if ($dispute->getDisputeId() === $this->getDisputeId()) {
-                return true;
+        if ($account) {
+
+            $viewableDisputes = $account->getAllDisputes();
+            foreach($viewableDisputes as $dispute) {
+                if ($dispute->getDisputeId() === $this->getDisputeId()) {
+                    return true;
+                }
             }
-        }
 
-        if ($this->getMediationState()->inMediation()) {
-            return (
-                $this->getMediationState()->getMediator()->getLoginId()        === $loginID ||
-                $this->getMediationState()->getMediationCentre()->getLoginId() === $loginID
-            );
+            if ($this->getMediationState()->inMediation()) {
+                return (
+                    $this->getMediationState()->getMediator()->getLoginId()        === $loginID ||
+                    $this->getMediationState()->getMediationCentre()->getLoginId() === $loginID
+                );
```

Comment:
Either extract these out to broken-down private functions, or a separate class, e.g. DisputeState